src/components/resumption/include/resumption/last_state.h

```diff
@@ -35,44 +35,26 @@
 
 #include <string>
 
-#include "utils/macro.h"
 #include "json/json.h"
 
 namespace resumption {
 
 class LastState {
  public:
   /**
-   * @brief Constructor
-   */
-  LastState(const std::string& app_storage_folder,
-            const std::string& app_info_storage);
-
-  /**
    * @brief Destructor
    */
-  ~LastState();
+  virtual ~LastState() {}
 
   /**
    * @brief Saving dictionary to filesystem
    */
-  void SaveToFileSystem();
-
-  /**
-   * @brief public dictionary
-   */
-  Json::Value dictionary;
-
- private:
-  DISALLOW_COPY_AND_ASSIGN(LastState);
+  virtual void SaveStateToFileSystem() = 0;
 
   /**
-   * @brief Load dictionary from filesystem
+   * @brief Get reference to dictionary
    */
-  void LoadFromFileSystem();
```

Comment:
@LuxoftAKutsan Well this method was in private scope of this class, so I decided to not include this method in interface of LastState class.